packages/replay/src/util/log.ts

```diff
@@ -11,22 +11,42 @@ export function logInfo(message: string, shouldAddBreadcrumb?: boolean): void {
 
   logger.info(message);
 
+  if (shouldAddBreadcrumb) {
+    addBreadcrumb(message);
+  }
+}
+
+/**
+ * Log a message, and add a breadcrumb in the next tick.
+ * This is necessary when the breadcrumb may be added before the replay is initialized.
+ */
+export function logInfoNextTick(message: string, shouldAddBreadcrumb?: boolean): void {
```

Comment:
I ended up splitting this into two methods `logInfo` and `logInfoNextTick`, as while the `log in next tick` change fixed some cases, but "broke" others (when logging right before a flush). So now we only explicitly log in the next tick in cases that could be logged before the replay is started.